utils/filter_streams.h

```diff
@@ -124,6 +135,19 @@ class filter_ostream : public std::ostream {
     std::map<std::string, std::string> options;
 };
 
+template<typename T>
+filter_ostream & operator << (filter_ostream & stream, T && data)
+{
+    try {
+        std::ostream & stdStream(stream);
+        stdStream << data;
```

Comment:
The universal reference is almost always used in conjunction with std::forward. In this case it should read `stdStream << std::forward(data);`